tests/PHPStan/Analyser/data/mb-convert-encoding-php8.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace MbConvertEncodingPHP8;
+
+/**
+ * @param 'foo'|'bar' $constantString
+ * @param array{foo: string, bar: int, baz: 'foo'} $structuredArray
+ * @param list<string> $stringList
+ * @param list<int> $intList
+ * @param 'foo'|'bar'|array{foo: string, bar: int, baz: 'foo'}|bool $union
+ */
+function test_mb_convert_encoding(
+	mixed $mixed,
+	string $constantString,
+	string $string,
+	array $mixedArray,
+	array $structuredArray,
+	array $stringList,
+	array $intList,
+	string|array|bool $union,
+	int $int,
+): void {
+	\PHPStan\Testing\assertType('array|string|false', mb_convert_encoding($mixed, 'UTF-8'));
+	\PHPStan\Testing\assertType('string', mb_convert_encoding($constantString, 'UTF-8'));
+	\PHPStan\Testing\assertType('string', mb_convert_encoding($string, 'UTF-8'));
+	\PHPStan\Testing\assertType('array', mb_convert_encoding($mixedArray, 'UTF-8'));
+	\PHPStan\Testing\assertType('array{foo: string, bar: int, baz: string}', mb_convert_encoding($structuredArray, 'UTF-8'));
+	\PHPStan\Testing\assertType('list<string>', mb_convert_encoding($stringList, 'UTF-8'));
+	\PHPStan\Testing\assertType('list<int>', mb_convert_encoding($intList, 'UTF-8'));
+	\PHPStan\Testing\assertType('array{foo: string, bar: int, baz: string}|string|false', mb_convert_encoding($union, 'UTF-8'));
+	\PHPStan\Testing\assertType('array|string|false', mb_convert_encoding($int, 'UTF-8'));
```

Comment:
should maybe be `string` (considering that `int` is cast to string by the union type) or an error (for strict types)